platform/feature-api/api/pod-api/src/api.ts

```diff
@@ -179,24 +179,30 @@ export interface PolyLifecycle {
  */
 export interface Endpoint {
     /**
-     * Perform a http post request via the endpoint in the pod
-     * @param body the necessary content of the call
-     * @returns an empty response or throws an error if something goes wrong
+     * Perform a http post request to a specific endpoint in the pod
+     * @param {string} endpointId - The ID of the endpoint to send the request to.
+     * @param {string} payload - The payload to send to the endpoint.
+     * @param {string} [contentType] - The content type of the payload.
+     * @param {string} [authToken] - The token to use for authentication. If not provided, the default
+     * token will be used.
      * @throws if an unsupported request goes through, if an endpoint is not reached or if a user denies a request
      */
-    send(
-        endpointId: string,
-        payload: string,
-        contentType?: string,
-        authToken?: string
-    ): Promise<void>;
+    send(request: {
+        endpointId: string;
+        payload: string;
+        contentType?: string;
+        authToken?: string;
+    }): Promise<void>;
 
     /**
-     * Perform a http get request via the endpoint in the pod
-     * @returns a promise with the response
+     * Performs an http GET request to the endpoint of the given ID.
+     * @param {string} endpointId - The ID of the endpoint.
+     * @param {string} [contentType] - The content type of the request.
+     * @param {string} [authToken] - The token to use for authentication.
+     * @returns a promise with the payload of the response
      * @throws if an unsupported request goes through, if an endpoint is not reached, if a user denies a request or if response is null
      */
-    get(endpointId: string, contentType?: string, authToken?: string): Promise<string>;
+    get(request: { endpointId: string; contentType?: string; authToken?: string }): Promise<string>;
```

Comment:
As said in the comment, this could be used to fulfill https://jira.polypoly.eu/browse/PROD4POD-1461. Have you considered, for instance, creating an interface that would include these three parameters? That way you avoid DRY, and also create an abstraction that can be used elsewhere.
Alternatively, a naming nit: calling both `request` may lead to confusion.

Reply:
That is correct and as said above we tried to go this route with endpointRequest and interface for that, but we changed it back because of the interface to the swift and android implementation. We would need to change everything there as well if you go that way and I would propose not to do that in this PR

Reply:
To communicate complex structures you probably need bubblewrap, which is in charge of encoding/decoding at either end, if I understood it correctly. That would be needed if a class is used, but I think that defining an interface is just what's been done now, only the structure is declared in a single place.